caterva2/services/sub.py

```diff
@@ -2266,7 +2344,7 @@ async def htmx_command(
     # Commands
     cmd = commands.get(argv[0])
     if cmd is not None:
-        if len(argv) != cmd.nargs:
+        if (cmd == ConcatCmd and len(argv) < 4) or (cmd != ConcatCmd and len(argv) != cmd.nargs):
```

Comment:
To avoid ending with too complex boolean expressions, I'd factor out the concat-specific code before.  Something like:

```python
if (cmd == ConcatCmd and len(argv) < 4):
    return htmx_error(request, f"Invalid syntax: expected {cmd.expected}")
if len(argv) != cmd.nargs:
    return htmx_error(request, f"Invalid syntax: expected {cmd.expected}")
```

which is is a little bit more verbose, but easier to read for your future self.